src/consensus/params.h

```diff
@@ -29,6 +29,11 @@ struct BIP9Deployment {
     int64_t nStartTime;
     /** Timeout/expiry MedianTime for the deployment attempt. */
     int64_t nTimeout;
+    /** If lock in occurs, delay activation until at least this block
+     *  height.  Note that activation will only occur on a retarget
+     *  boundary.
+     */
+    int min_activation_height{0};
```

Comment:
Would it make sense to always set `min_activation_height` to a retarget boundary (and assert that)?

Reply:
If it's not on a retarget boundary, it will be treated as if it were at the next higher retarget boundary; so would be misleading but not actually buggy. Added a sanity check to versionbits_tests.